homeassistant/components/axis/__init__.py

```diff
@@ -64,3 +78,19 @@
     }
 
     hass.config_entries.async_update_entry(config_entry, options=options)
+
+
+async def async_migrate_entry(hass, config_entry):
+    """Migrate old entry."""
+    LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    if config_entry.version == 1:
+        device = config_entry.data.pop(CONF_DEVICE)
```

Comment:
Migrations are annoying because it means people cannot rollback versions. Is it really necessary to flatten the device object here ?

Reply:
I could keep the old data intact to support downgrade. Better?

Reply:
I would prefer that. We should only do migrations if it's really necessary. I don't think this one falls in that category.

Reply:
Done